### PR TITLE
lbl: more commands

### PR DESCRIPTION
**Pull request by HookedBehemoth**
I arranged the existing commands by id.

### CODE REVIEW

**Comment by yellows8** [comment]
Some of these are used internally by am, mention that in the docs for the relevant cmds (see wiki).

File: nx/include/switch/services/lbl.h [hunk 1]
```diff
@@ -17,15 +17,37 @@ void lblExit(void);
...
+Result lblIsDimmingEnabled(bool *out_value);
+
 Result lblEnableAutoBrightnessControl(void);
 Result lblDisableAutoBrightnessControl(void);
 Result lblIsAutoBrightnessControlEnabled(bool *out_value);
+
+Result lblSetAmbientLightSensorValue(float value);
+Result lblGetAmbientLightSensorValue(bool *over_limit, float *lux);
+
+/* Only available on [3.0.0+] */
```
Comment (yellows8) [inline nx/include/switch/services/lbl.h]: Doc the required sysver for each individual cmd (also, period).

File: nx/source/services/lbl.c [hunk 2]
```diff
@@ -37,6 +38,37 @@ static Result _lblCmdNoInOutBool(bool *out, u32 cmd_id) {
     return rc;
 }
 
+static Result _lblCmdNoInOutFloat(float *out, u32 cmd_id) {
+    float tmpout=0;
+    Result rc = serviceDispatchOut(&g_lblSrv, cmd_id, tmpout);
```
Comment (yellows8) [inline nx/source/services/lbl.c]: Just use *out directly.

File: nx/source/services/lbl.c [hunk 3]
```diff
@@ -45,14 +77,26 @@ Result lblSwitchBacklightOff(u64 fade_time) {
     return _lblCmdInU64NoOut(fade_time, 7);
 }
 
-Result lblSetCurrentBrightnessSetting(float brightness) {
-    return serviceDispatchIn(&g_lblSrv, 2, brightness);
+Result lblGetBacklightSwitchStatus(bool *out_value) {
+    u32 tmpout=0;
+    Result rc = serviceDispatchOut(&g_lblSrv, 8, tmpout);
+    if (R_SUCCEEDED(rc) && out_value) *out_value = tmpout & 0xff;
```
Comment (yellows8) [inline nx/source/services/lbl.c]: Is u32->bool really correct?

File: nx/source/services/lbl.c [hunk 4]
```diff
@@ -45,14 +77,26 @@ Result lblSwitchBacklightOff(u64 fade_time) {
...
+
+Result lblDisableDimming(void) {
+    return _lblCmdNoIO(10);
 }
 
+Result lblIsDimmingEnabled(bool *out_value) {
+    return _lblCmdNoInOutBool(out_value, 11);
+}
+
+
```
Comment (yellows8) [inline nx/source/services/lbl.c]: Remove extra newline.

File: nx/source/services/lbl.c [hunk 5]
```diff
@@ -64,3 +108,55 @@ Result lblDisableAutoBrightnessControl(void) {
...
+    Result rc = serviceDispatchOut(&g_lblSrv, 16, out);
+    if (R_SUCCEEDED(rc)) {
+        if (over_limit) *over_limit = out.over_limit & 0xff;
+        if (lux) *lux = out.lux;
+    }
+    return rc;
+}
+
+Result lblIsAmbientLightSensorAvailable(bool *out_value) {
```
Comment (yellows8) [inline nx/source/services/lbl.c]: Uses _lblCmdNoIO, but has output param...

File: nx/source/services/lbl.c (hunk 3 above)
Comment (HookedBehemoth) [reply nx/source/services/lbl.c]: You were right.
I'll add an enum.
0 disabled, 1 enabled, 2, enabling, 3 disabling

File: nx/include/switch/services/lbl.h [hunk 6]
```diff
@@ -17,15 +24,62 @@ void lblExit(void);
...
+Result lblSetCurrentBrightnessSettingForVrMode(float brightness);
+
+/**
+ * @note Only available on [3.0.0+].
+ */
+Result lblGetCurrentBrightnessSettingForVrMode(float *out_value);
+
+/**
+ * @note Only available on [3.0.0+].
+ * @note Used internally by am SetVrModeEnabled.
```
Comment (yellows8) [inline nx/include/switch/services/lbl.h]: `by \ref applet{name}.`, same below.

File: nx/include/switch/services/lbl.h [hunk 7]
```diff
@@ -17,15 +24,62 @@ void lblExit(void);
...
+Result lblEnableDimming(void);
+Result lblDisableDimming(void);
+Result lblIsDimmingEnabled(bool *out_value);
+
 Result lblEnableAutoBrightnessControl(void);
 Result lblDisableAutoBrightnessControl(void);
 Result lblIsAutoBrightnessControlEnabled(bool *out_value);
+
+Result lblSetAmbientLightSensorValue(float value);
+Result lblGetAmbientLightSensorValue(bool *over_limit, float *lux);
```
Comment (yellows8) [inline nx/include/switch/services/lbl.h]: applet uses this and lblIsAmbientLightSensorAvailable, add that to these docs (see wiki).

File: nx/include/switch/services/lbl.h [hunk 8]
```diff
@@ -8,6 +8,13 @@
 #include "../types.h"
 #include "../sf/service.h"
 
+typedef enum {
+    LblBacklightSwitchStatus_Disabled = 0,
```
Comment (yellows8) [inline nx/include/switch/services/lbl.h]: Align `=` with each line.

File: nx/source/services/lbl.c [hunk 9]
```diff
@@ -37,6 +38,34 @@ static Result _lblCmdNoInOutBool(bool *out, u32 cmd_id) {
     return rc;
 }
 
+static Result _lblCmdNoInOutFloat(float *out, u32 cmd_id) {
```
Comment (yellows8) [inline nx/source/services/lbl.c]: Add an input float version of this, for cmds which need it?

File: nx/include/switch/services/lbl.h [hunk 10]
```diff
@@ -8,6 +8,13 @@
 #include "../types.h"
 #include "../sf/service.h"
 
+typedef enum {
+    LblBacklightSwitchStatus_Disabled  = 0,
+    LblBacklightSwitchStatus_Enabled   = 1,
+    LblBacklightSwitchStatus_Enabling  = 2,
+    LblBacklightSwitchStatus_Disabling = 3
```
Comment (fincs) [inline nx/include/switch/services/lbl.h]: Missing comma after last enum member

**Pull request merged**